lib/Frontend/nvqpp/ConvertExpr.cpp

```diff
@@ -1083,18 +1083,36 @@ bool QuakeBridgeVisitor::TraverseLambdaExpr(clang::LambdaExpr *x,
 
 bool QuakeBridgeVisitor::TraverseMemberExpr(clang::MemberExpr *x,
                                             DataRecursionQueue *) {
-  [[maybe_unused]] auto typeStackDepth = typeStack.size();
-  // FIXME!
-  // Accessing data members is not currently supported. For function members, we
-  // want to push the type of the function, since the visit to CallExpr requires
-  // a type to have been pushed.
-  if (auto *methodDecl = dyn_cast<clang::CXXMethodDecl>(x->getMemberDecl()))
+  if (auto *methodDecl = dyn_cast<clang::CXXMethodDecl>(x->getMemberDecl())) {
+    // For function members, we want to push the type of the function, since the
+    // visit to CallExpr requires a type to have been pushed.
+    [[maybe_unused]] auto typeStackDepth = typeStack.size();
     if (!TraverseType(methodDecl->getType()))
       return false;
-  assert(typeStack.size() == typeStackDepth + 1);
+    assert(typeStack.size() == typeStackDepth + 1);
+  }
+  if (auto *field = dyn_cast<clang::FieldDecl>(x->getMemberDecl())) {
+    [[maybe_unused]] auto typeStackDepth = typeStack.size();
+    if (!TraverseType(field->getType()))
+      return false;
+    assert(typeStack.size() == typeStackDepth + 1);
+  }
   return Base::TraverseMemberExpr(x);
 }
 
+bool QuakeBridgeVisitor::VisitMemberExpr(clang::MemberExpr *x) {
+  if (auto *field = dyn_cast<clang::FieldDecl>(x->getMemberDecl())) {
+    auto loc = toLocation(x->getSourceRange());
+    auto object = popValue(); // DeclRefExpr
+    std::int32_t offset = field->getFieldIndex();
+    auto ty = popType();
+    return pushValue(builder.create<cc::ComputePtrOp>(
+        loc, cc::PointerType::get(ty), object,
+        SmallVector<cc::ComputePtrArg>{0, offset}));
+  }
+  return true;
```

Comment:
Do we want to throw an error if it's not a `FieldDecl`? I.e. if it's a `VarDecl` (for static data members), a `CXXMethodDecl`, or an `EnumConstantDecl`?

Reply:
Maybe. I can try it. There are still cases where we may traverse C++ code that we don't really process intentionally. Consider that we're visiting post-order, so it is possible that we visit subtrees that then are elided when the parent is visited and recognized as a something we want to translate to Quake or CC more directly.

Reply:
It doesn't look like error messages is the way to go here.